packages/compartment-mapper/test/custom-parser.test.js

```diff
@@ -0,0 +1,67 @@
+import 'ses';
+import fs from 'fs';
+import test from 'ava';
+import url from 'url';
+import { loadLocation } from '../src/import.js';
+import { makeReadPowers } from '../src/node-powers.js';
+
+const { freeze } = Object;
+
+const readPowers = makeReadPowers({ fs, url });
+const { read } = readPowers;
+
+test('defining a custom parser works', async t => {
+  const fixture = new URL(
+    'fixtures-0/node_modules/markdown/README.md',
+    import.meta.url,
+  ).toString();
+
+  const application = await loadLocation(read, fixture, {
+    parsers: [
+      {
+        parser: {
+          // This parser parses markdown files. Use your imagination
+          parse: async (
+            bytes,
+            specifier,
+            moduleLocation,
+            packageLocation,
+            { compartmentDescriptor },
+          ) => {
+            const execute = moduleEnvironmentRecord => {
+              moduleEnvironmentRecord.default = `${bytes}`;
+            };
+
+            t.truthy(
+              compartmentDescriptor,
+              'compartmentDescriptor is passed to parser',
+            );
```

Comment:
I could use a demonstration of a behavior that would switch on the compartment descriptor.

Reply:
Think policy enforcement - the parser is responsible for the security or the confinement of what it's processing and custom extension to policy could configure its behavior.

Reply:
Yes, that's one of the test cases I haven't implemented yet.